Implement BaseConverter UI with Toggle Layout Support (BTMC / TERN) - Feature/form-jayanthUtukuri

jayanth2909 opened the pull request:
**Summary**  
This PR implements the main form UI using custom WPF controls as per the Figma specifications. The application supports layout switching between **BTMC** and **TERN** via toggle buttons in `Main.xaml`, and includes reusable custom controls with proper styling, behavior, and placeholder logic.

---

**Changes Implemented**

- Integrated BTMC and TERN layout switching logic directly into `Main.xaml` and `Main.xaml.cs`
- Created and styled:
  - `CustomTextBox` with built-in placeholder support (no external behavior class)
  - `CustomComboBox` with CornerRadius and custom styling
  - Custom `Clear` button to reset input/output fields and restore placeholders
  - Custom `Swap` button to intelligently swap input/output while preserving placeholders
- Applied consistent theme: corner radius, color palette, font family
- Implemented `TabControl` and `TabItem` styles to match pill toggle layout
- Refactored and organized code to match folder structure and maintainability goals
- Resolved merge conflict in `Readme.md`

---

**How to Test the Changes**

1. Check out the branch:

    ```bash
    git fetch origin
    git checkout feature/form-jayanthUtukuri
    ```

2. Open the solution in Visual Studio and run the app.
3. Validate the following:

- The main window loads with BTMC visible by default.
-Toggle buttons at the top switch between BTMC and TERN views.
-`CustomComboBox` controls are styled and selectable.
-`CustomTextBox` controls show placeholder text that disappears on focus and reappears on blur.
-**Clear** button resets all visible input fields correctly.
-**Swap** button exchanges input/output values and restores placeholders if needed.
-All controls respect spacing, font, and rounded border specifications from the Figma design.

---

**Expected Outcome**

- The UI renders correctly with BTMC/TERN switching behavior.
- All custom controls behave as expected and maintain placeholder functionality.
- Layout and styles follow the design system guidelines.
- The solution structure is modular, clean, and ready for logic integration.

---
Changes made on 05/23/2025
1. Implemented input validation logic in BTMC.xaml.cs
2. Made the toggle container generic and reusable via styled TabControl.


@ternksym ready for your review. Let me know if any changes are required. Thanks!

M20A23X commented:
@jayanth2909 The overall look is good. There are some mentions:
- the inputs act weird - the placeholder should disappear when a user starts typing:
![image](https://github.com/user-attachments/assets/0c06b570-65bc-448b-a924-80c326eb029f)
- I checked the code, and it's not organized at all - there is a structure of folders prepared for different parts of the app: like theme, converters (it's not about the form - it's about [the other thing](https://learn.microsoft.com/en-us/dotnet/api/system.converter-2?view=net-9.0)), custom components. **You need to follow this structure, because if the code works (it's only a one of the steps), but not organized well - the other devs will have a lot of troubles exploring and reorganizing it and you need to pay attention to it, because it's a critical thing**

jayanth2909 commented:
Hi maksym,

Thank you for reverting back with the errors I have done.
1) About the first error placeHolder doesn’t disappear as soon as you place the cursor- this I haven’t implemented logically as I thought only UI should be implemented in this task but thanks for letting me know I’ll do it and push the changes 
 
 2)  I am new to XAML although I have knowledge working on C# but again thanks for letting me know the errors I have made. I’ll make this changes by EOD and push it back to the repository.

Thanks & Regards,
Jayanth utukuri

M20A23X commented:
@jayanth2909 could you try to implement it pls ?

jayanth2909 commented:
Hey maksym ,
I have refactored all the code and I just need to implement the placeholder part logically and I make sure I’ll push it today by EOD. Sorry for the delay.

jayanth2909 commented:
Hey Maksym,
I have made the changes you have told. can you please check and revert back to me

jayanth2909 commented:
Hey Maksym,

I’ve completed all the changes related to the TabItem style. The toggle container is now implemented in a generic way—so if you add additional TabItems, the rounded styles will automatically apply to all of them.

Additionally, I’ve implemented input validation for the BTMC layout, and the dropdown options are now dynamically sourced from the ALU folder.
